test/injected-script-test.js

```diff
@@ -1,1033 +1,1359 @@
+				}
+			}
+
+			customElements.define("circular-app-2", CircularApp);
+
+			const el = new CircularApp().initialize();
+			const { viewModelData, typeNames, messages } = devtools.getViewModelData(
+				el
+			).detail;
+
+			assert.deepEqual(
+				viewModelData,
+				{ circular: {} },
+				"gets empty object for circular property"
+			);
+
+			assert.deepEqual(
+				typeNames,
+				{ circular: "Object{}" },
+				"gets correct name for circular property"
+			);
+
+			assert.equal(typeof messages, "object");
+			assert.equal(typeof messages.circular, "object");
+			assert.equal(messages.circular.type, "error");
+			assert.ok(
+				messages.circular.message.match(/Error getting value of "circular":/)
+			);
+		});
+
+		it("can handle infinite recursion (#46)", () => {
+			class Thing extends ObservableObject {
+				static get props() {
+					return {
+						anotherThing: {
+							default() {
+								return new Thing();
+							}
+						}
+					};
+				}
+			}
+
+			class C extends StacheElement {
+				static get view() {
+					return "<p>hello</p>";
+				}
+
+				static get props() {
+					return {
+						thing: {
+							get default() {
+								return new Thing();
+							}
+						}
+					};
+				}
+			}
+
+			customElements.define("circular-app", C);
+
+			const el = new C().initialize();
+			const { viewModelData, typeNames } = devtools.getViewModelData(el).detail;
+
+			assert.deepEqual(
+				viewModelData,
+				{ thing: {} },
+				"gets empty object for recursive property"
+			);
+
+			assert.deepEqual(typeNames, { thing: "Thing{}" }, "gets correct name");
+		});
+
+		it("can handle nulls and undefineds", () => {
+			class C extends StacheElement {
+				static get view() {
+					return "<p>hello</p>";
+				}
+
+				static get props() {
+					return {
+						thisIsNull: { default: null, type: type.maybe(String) },
+						thisIsUndefined: { default: undefined, type: type.maybe(String) }
+					};
+				}
+			}
+
+			customElements.define("app-with-nulls", C);
+
+			const el = new C().initialize();
+
+			const { viewModelData, messages, undefineds } = devtools.getViewModelData(
+				el
+			).detail;
+
+			assert.deepEqual(
+				viewModelData,
+				{ thisIsNull: null, thisIsUndefined: undefined },
+				"gets null for property that is null and undefined for property that is undefined"
+			);
+
+			assert.deepEqual(
+				undefineds,
+				["thisIsUndefined"],
+				"gets undefineds array with paths whose value is `undefined`"
+			);
+
+			assert.deepEqual(messages, {}, "gets no messages");
+		});
+
+		it("can handle empty Maps / Lists / Objects / arrays", () => {
+			class C extends StacheElement {
+				static get view() {
+					return "<p>hello</p>";
+				}
+
+				static get props() {
+					return {
+						emptyMap: {
+							get default() {
+								return new ObservableObject();
+							}
+						},
+						emptyList: {
+							get default() {
+								return new ObservableArray();
+							}
+						},
+						emptyObject: {
+							type: type.Any,
```

Comment:
I think these `type.Any`s can be removed. I think this was just to prevent DefineMap from converting these to observables.

Reply:
Removed.